### ISSUE TITLE
SpellerHighlightLayer uses hard coded spell check squiggle color

### ISSUE THREAD

**Issue by elyoh** [dotnet/wpf#1519]
The [spell check underline squiggle](https://github.com/dotnet/wpf/blob/master/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Documents/SpellerHighlightLayer.cs#L153) uses a hard coded color.  This behaviour is undesirable as it can lead to accessibility problems for users with color vision deficiency and when using custom themes.

For example, when the red squiggle is displayed on a black background (if using the HC Black theme) [insufficient color contrast is predicted](https://accessibilityinsights.io/) for users with Red-Weak/Protanomaly (and Green-Weak/Deuteranomaly). 

Original view
![Capture](https://user-images.githubusercontent.com/14054078/62410535-bd9c1200-b5de-11e9-9403-ec0c4f15931a.PNG)

[Red-Weak/Protanomaly simulation](https://www.color-blindness.com/coblis-color-blindness-simulator/)
![redweak](https://user-images.githubusercontent.com/14054078/62410538-c0970280-b5de-11e9-9fc4-27bed00d84ee.PNG)
![redweakmax](https://user-images.githubusercontent.com/14054078/62410656-8b8baf80-b5e0-11e9-83bc-e100dd6b68f9.PNG)

Ideally it would be possible to override this color from controls which support spell check.

**Comment by anjalisheel-wpf** [comment]
@elyoh It would be extremely helpful if you can provide us with some sample WPF code, then it will be easier for us to reproduce, investigate and fix it.

**Comment by elyoh** [comment]
I can provide a sample app, but it is nothing more than a hello world WPF app with the following in the main window:

```
<StackPanel>
    <TextBox SpellCheck.IsEnabled="True" Name="myTextBox">Here is a note with a speling mistake.</TextBox>
</StackPanel>
```
The spell check squiggle displayed in this case is always red.  This is hard coded and cannot be changed:
https://github.com/dotnet/wpf/blob/cf2a8903f9569239476b38c57058c892cabd9e8e/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Documents/SpellerHighlightLayer.cs#L153

There are many scenarios when this is not desrirable.

If you run such an app on Windows 10 and earlier, with the high contrast black theme, you have a red squiggle on a black background (first image in issue).  This is underirable for users with Red-Green colour vision deficiency.  If you take a screen shot of the main window and run it through a simulator of what someone with Red-Green colour vision deficiency would see, then Accessibility Inights shows insufficient contrast between the "red" squiggle and the black background.  I used this simulator for the to generate the second image in the original issue: https://www.color-blindness.com/coblis-color-blindness-simulator/

Further, if the TextBox uses a custom Background or Forground colour, developers cannot ensure that the red squiggle will be clearly visible to all users, again, due to the hard coded squiggle colour.
